cmd/tsidp/tsidp.go

```diff
@@ -90,6 +95,38 @@ func main() {
 		if !anySuccess {
 			log.Fatalf("failed to listen on any of %v", st.TailscaleIPs)
 		}
+		sc, err := lc.GetServeConfig(ctx)
+		if err != nil {
+			log.Fatalf("could not get serve config: %v", err)
+		}
+		if sc == nil {
+			sc = new(ipn.ServeConfig)
+		}
+		watcher, err = lc.WatchIPNBus(ctx, ipn.NotifyInitialState|ipn.NotifyNoPrivateKeys)
+		if err != nil {
+			log.Fatalf("could not set up ipn bus watcher: %v", err)
+		}
+		defer watcher.Close()
+		n, err := watcher.Next()
+		if err != nil {
+			log.Fatalf("could not get initial state from ipn bus watcher: %v", err)
+		}
+		if n.SessionID == "" {
+			log.Fatalf("missing sessionID in ipn.Notify")
+		}
+		foregroundSc := new(ipn.ServeConfig)
+		mak.Set(&sc.Foreground, n.SessionID, foregroundSc)
+		serverURL := strings.TrimSuffix(st.Self.DNSName, ".")
+		fmt.Printf("setting funnel for %s:%v\n", serverURL, uint16(*flagPort))
+		foregroundSc.SetFunnel(serverURL, uint16(*flagPort), *flagFunnel)
+		mak.Set(&foregroundSc.TCP, uint16(*flagPort), &ipn.TCPPortHandler{
+			// TODO(naman): choose ips better?
+			TCPForward: net.JoinHostPort(st.TailscaleIPs[0].String(), portStr),
+		})
+		err = lc.SetServeConfig(ctx, sc)
```

Comment:
This is all pretty low-level. What's the goal here, as opposed to just using ListenFunnel? It could use some comments if this is necessary, but I'm not sure why it would be.

Reply:
This is all for the mode where we use the local tailscaled instead of tsnet. That mode works by simply attaching standard TCP listeners to the Tailscale IPs assigned to the local tailscaled, so to funnel those out we have to do the equivalent of the `tailscale funnel` CLI command here. I can add comments to that effect.

Also, @maisem informed me that this PR is missing some things, so marking it as a draft for now.